sql/core/src/test/scala/org/apache/spark/sql/execution/bucketing/CoalesceBucketsInJoinSuite.scala

```diff
@@ -103,46 +119,69 @@ class CoalesceBucketsInSortMergeJoinSuite extends SQLTestUtils with SharedSparkS
   }
 
   test("bucket coalescing - basic") {
-    withSQLConf(SQLConf.COALESCE_BUCKETS_IN_SORT_MERGE_JOIN_ENABLED.key -> "true") {
+    withSQLConf(SQLConf.COALESCE_BUCKETS_IN_JOIN_ENABLED.key -> "true") {
+      run(JoinSetting(
+        RelationSetting(4, None), RelationSetting(8, Some(4)), joinOperator = sortMergeJoin))
+      run(JoinSetting(
+        RelationSetting(4, None), RelationSetting(8, Some(4)), joinOperator = shuffledHashJoin,
+        shjBuildSide = Some(BuildLeft)))
+      // Coalescing bucket should not happen when the target is on shuffled hash join
```

Comment:
How about pulling this out as a separate test?

Reply:
@imback82 - yes, extracting this to a new test - `bucket coalescing shouldn't be applied to shuffled hash join build side`.